app/services/next_references/next_transaction_reference.service.js

```diff
@@ -8,66 +8,115 @@ const { SequenceCounterModel } = require('../../models')
-   * - `1` is a fixed digit "1"
-   * - `999999` is our sequential transaction number padded to a 6-digit numeric string
-   *
-   * For example, if the region was 'R', the next transaction number was 3, and it was for a credit note the reference
-   * would be `RAC1000003`.
-   *
-   * If an invalid region & regime pair is supplied, an Objection `NotFoundError` is thrown
+   * If an invalid region & regime pair is supplied, an Objection `NotFoundError` is thrown. If an unvalid ruleset is
+   * supplied a `TypeError` is thrown.
    *
    * @param {string} regimeId Id of the regime to get the next reference for
    * @param {string} region The region to get the next reference for
+   * @param {string} ruleset The ruleset to get the next reference for
    * @param {string} transactionType Either a `'C'` or an `'I'` which denotes whether the invoice the reference is for
    * is an invoice or a credit note.
    * @param {Object} [trx] Optional Objection database `transaction` object to be used in the update to
    * `sequence_counters`
    *
    * @returns {string} the generated transaction reference
    */
-  static async go (regimeId, region, transactionType, trx = null) {
-    const result = await this._updateSequenceCounter(regimeId, region, trx)
+  static async go (regimeId, region, ruleset, transactionType, trx = null) {
+    const result = await this._updateSequenceCounter(regimeId, region, ruleset, trx)
 
-    return this._response(region, result.transactionNumber, transactionType)
+    return this._response(region, ruleset, transactionType, result)
   }
 
-  static async _updateSequenceCounter (regimeId, region, trx) {
-    return SequenceCounterModel.query(trx)
+  static async _updateSequenceCounter (regimeId, region, ruleset, trx) {
+    const rulesetColumnNames = this._rulesetColumnNames()[ruleset]
+
+    const result = await SequenceCounterModel.query(trx)
       .findOne({
         regime_id: regimeId,
         region
       })
-      .increment('transaction_number', 1)
-      .returning('transaction_number')
+      .increment(rulesetColumnNames.raw, 1)
+      .returning(rulesetColumnNames.raw)
       .throwIfNotFound({
         message: 'Invalid combination of regime and region'
       })
+
+    return result[rulesetColumnNames.camel]
+  }
+
+  static _rulesetColumnNames () {
+    return {
+      presroc: {
+        raw: 'transaction_number_presroc', camel: 'transactionNumberPresroc'
+      },
+      sroc: {
+        raw: 'transaction_number_sroc', camel: 'transactionNumberSroc'
+      }
+    }
   }
 
-  static _response (region, transactionNumber, transactionType) {
-    return `${region}A${transactionType}1${this._padNumber(transactionNumber)}`
+  static _response (region, ruleset, transactionType, transactionNumber) {
+    if (ruleset === 'presroc') {
+      return this._presrocResponse(region, transactionType, transactionNumber)
+    } else {
+      return this._srocResponse(region, transactionType, transactionNumber)
+    }
```

Comment:
We could drop the `else` clause here if we wanted? Unless keeping it would make the intent clearer?

```suggestion
    if (ruleset === 'presroc') {
      return this._presrocResponse(region, transactionType, transactionNumber)
    }

    return this._srocResponse(region, transactionType, transactionNumber)
```